### PR TITLE
N orient 3

### PR DESCRIPTION
**Pull request by AliHashemi-ai**
Hi @agramfort  

I have imported the bcs function from mne to test n_orient = 3 in a correct way as we discussed 
I think we need to go through the code to see whether I did it in a correct way?

### CODE REVIEW

**Comment by AliHashemi-ai** [comment]
Hi @agramfort  The code currently passes all the tests  except iterative_sqrt
It should be something related to the initialization of the alpha since previously it worked well with the following manual tuning of alpha:  https://github.com/braindatalab/BSI-Zoo/blob/2eff796807882a3aa3fc7aa70940f296d8363f22/bsi_zoo/estimators.py#L229

    alpha_max = abs(L.T.dot(y)).max() / len(L)
    alpha = alpha * alpha_max

(I have commented out these two lines) 

But right now we have this in the bcd code 
alpha_lc = alpha / lipschitz_constant

which I think change the scale of the alpha a little bit. would be appreciate if you could take a look at this. (currently all the test are passed since I commented out iterative_sqrt function in the test)

**Comment by agramfort** [comment]
@AliHashemi-ai can you see why CIs are red here? thanks

**Comment by AliHashemi-ai** [comment]
> @AliHashemi-ai can you see why CIs are red here? thanks

I have checked that this is the error: 

image

do you have any idea how to resolve it @agramfort ? it seems it refers to an whitespace around an operator

**Comment by agramfort** [comment]
@AliHashemi-ai run flake8 locally on your code to replicate the problem and fix it

**Comment by AliHashemi-ai** [comment]
> @AliHashemi-ai run flake8 locally on your code to replicate the problem and fix it

Thanks @agramfort . Done. Now the flake8 error is fixed. Looking forward to your comment on how to tune alpha for iterative_sqrt function. Please look at my explanations above.

**Comment by agramfort** [comment]
@AliHashemi-ai it should be ok now. Basically with sqrt you have weights which are zeros and it breaks the solvers. So the fix is to screen dipoles with zero weights and not to consider them in the inner loop

File: bsi_zoo/tests/test_estimators.py [hunk 1]
```diff
@@ -15,7 +15,6 @@
 )
 
 
-# TODO: make the iterative type-II methods more efficient
 @pytest.mark.parametrize("n_times", [5])
 # @pytest.mark.parametrize("orientation_type", ["fixed", "free"])
 @pytest.mark.parametrize("orientation_type", ["free"])
```
Comment (agramfort) [inline bsi_zoo/tests/test_estimators.py:20]: green @AliHashemi-ai !

now can we put back the fixed in the test too?

Comment (AliHashemi-ai) [reply bsi_zoo/tests/test_estimators.py:20]: Sure @agramfort Let's do this. Shall I do it in this branch?

Comment (agramfort) [reply bsi_zoo/tests/test_estimators.py:20]: yes try and if it's more complex than expected we'll do another PR.

Comment (AliHashemi-ai) [reply bsi_zoo/tests/test_estimators.py]: > yes try and if it's more complex than expected we'll do another PR.

I change the code here https://github.com/braindatalab/BSI-Zoo/blob/n_orient_3/bsi_zoo/estimators.py#L61 and now all the tests for fixed and free orientation are passed

**Comment by AliHashemi-ai** [comment]
feel free to merge it with master if you agree with the this version

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
we did it @AliHashemi-ai 🎉 👏 

let's move on to the next item of our todo list now

**Pull request merged**